server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -342,8 +319,9 @@ public void onFailure(Exception e) {
                         ),
                         e
                     );
-                    // since we are not able to update the status, we will clear the voting config exclusion we have set earlier
-                    clearVotingConfigExclusionAndUpdateStatus(false, false);
+                    // TODO - this might not be needed
+                    // will go ahead and clear the voting config and mark the status as failed
+                    decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.FAILED, statusUpdateListener());
```

Comment:
why do you say it is not needed?
You need to move it to Failed status so that next time decommissioning request can be retried.

Reply:
This is listener's onFailure to update the decommission status to `DRAINING`. Now that we know the status update call failed, can we not safely assume that status update call to mark status as `FAILED` will also fail? Although there is no harm in sending a status update call, but if we are certain that it would fail, should we even send that task?

Reply:
Attempting a failed call anyways. Added a comment for reference - 
```
// This decommission state update call will most likely fail as the state update call to 'DRAINING' failed. But attempting it anyways as FAILED update might still pass as it doesn't have dependency on the current state
```